service/src/main/java/bio/terra/catalog/iam/SamAuthenticatedUserRequestFactory.java

```diff
@@ -2,33 +2,36 @@
 
 import bio.terra.common.exception.UnauthorizedException;
 import bio.terra.common.iam.AuthenticatedUserRequest;
-import bio.terra.common.iam.AuthenticatedUserRequestFactory;
 import bio.terra.common.iam.BearerTokenParser;
 import javax.servlet.http.HttpServletRequest;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import org.springframework.web.context.annotation.SessionScope;
 
 /**
- * An {@link AuthenticatedUserRequestFactory} which always resolves the user email and subjectId
- * from Sam given a request token.
+ * A Component which always resolves the user email and subjectId from Sam given a request token.
  *
  * <p>This is important for calls made by pet service accounts, which will have a pet email in the
  * request header, but Sam will return the owner's email.
  */
 @Component
-public class SamAuthenticatedUserRequestFactory implements AuthenticatedUserRequestFactory {
-  private static final String OAUTH2_ACCESS_TOKEN = "OAUTH2_CLAIM_access_token";
-  private static final String AUTHORIZATION = "Authorization";
-  private final SamService samService;
+@SessionScope
+public class SamAuthenticatedUserRequestFactory {
+  static final String OAUTH2_ACCESS_TOKEN = "OAUTH2_CLAIM_access_token";
+  static final String AUTHORIZATION = "Authorization";
+  private final SamUserStatusService samService;
+  private final HttpServletRequest request;
+  private AuthenticatedUserRequest cachedUser;
```

Comment:
> What expires the cache/when does the user expire? And how is it cached specifically, is it based on accessToken? Or request context?

The lifetime is tied to the containing object (`SamAuthenticatedUserRequestFactory`) so it's is determined by the scope ( of that object. Since it has `@SessionScope`, its lifetime is tied to the user's browser session. I think session is the correct scope to use, but the other HTTP scope to use is `RequestScope`. That won't cache the value across multiple requests, but will cache it within a single HTTP request operation.

If you're not familiar with bean scopes, check out https://docs.spring.io/spring-framework/docs/current-SNAPSHOT/reference/html/core.html#beans-factory-scopes

Reply:
I've changed the code to use request scope instead, since our app doesn't have sessions. This is "safe" in that spring will throw an error if a component tries to use a request scope object outside of a request scope. Although right now everything our app does is only through request scoped operations.